Tools/ViewsEstimators.py

```diff
@@ -6,118 +6,333 @@
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
+from sklearn.dummy import DummyClassifier
+
+from xgboost import XGBRegressor, XGBClassifier, XGBRFRegressor, XGBRFClassifier
+from lightgbm import LGBMClassifier, LGBMRegressor
 
 
+####################
+# HurdleRegression #
+####################
 class HurdleRegression(BaseEstimator):
     """ Regression model which handles excessive zeros by fitting a two-part model and combining predictions:
             1) binary classifier
             2) continuous regression
     Implementeted as a valid sklearn estimator, so it can be used in pipelines and GridSearch objects.
     Args:
-        clf_name: currently supports either 'logistic' or 'LGBMClassifier'
-        reg_name: currently supports either 'linear' or 'LGBMRegressor'
+        clf_name: name of a classifier sub-model to use
+        reg_name: name of a regression sub-model to use
         clf_params: dict of parameters to pass to classifier sub-model when initialized
         reg_params: dict of parameters to pass to regression sub-model when initialized
     """
-
+    # Define the constructor method for the class
     def __init__(self,
                  clf_name: str = 'logistic',
                  reg_name: str = 'linear',
                  clf_params: Optional[dict] = None,
                  reg_params: Optional[dict] = None):
-
+        
         self.clf_name = clf_name
         self.reg_name = reg_name
         self.clf_params = clf_params
         self.reg_params = reg_params
-        self.clf_fi = []
-        self.reg_fi = []
-
+        
     @staticmethod
-    def _resolve_estimator(func_name: str):
+    def _resolve_estimator(estimator_name: str, random_state=None):
         """ Lookup table for supported estimators.
         This is necessary because sklearn estimator default arguments
         must pass equality test, and instantiated sub-estimators are not equal. """
 
-        funcs = {'linear': LinearRegression(),
-                 'logistic': LogisticRegression(solver='liblinear'),
-                 'LGBMRegressor': LGBMRegressor(n_estimators=250),
-                 'LGBMClassifier': LGBMClassifier(n_estimators=250),
-                 'RFRegressor': XGBRFRegressor(n_estimators=250,n_jobs=-2),
-                 'RFClassifier': XGBRFClassifier(n_estimators=250,n_jobs=-2),
-                 'GBMRegressor': GradientBoostingRegressor(n_estimators=200),
-                 'GBMClassifier': GradientBoostingClassifier(n_estimators=200),
-                 'XGBRegressor': XGBRegressor(n_estimators=100,learning_rate=0.05,n_jobs=-2),
-                 'XGBClassifier': XGBClassifier(n_estimators=100,learning_rate=0.05,n_jobs=-2),
-                 'HGBRegressor': HistGradientBoostingRegressor(max_iter=200),
-                 'HGBClassifier': HistGradientBoostingClassifier(max_iter=200),
+        estimators = {
+                    'linear': LinearRegression(),
```

Comment:
Why is n_jobs removed?

Also. it is unfortunate that so many hyperparameters (HPs) are hardcoded here. They would be in the config file, but I do realize that since there are multiple different models to choose from, this could get messy. Still, we need to find a way. This is important so that when we create a sweep file for W&B we can iterate over both different HPs and models.

Also, the random_state should also be defined in the config

Reply:
Agree 100%. This is something we should agree on how to implement and follow the agreement. I do not want to create more technical debt just doing it the way I want if others do it in a different way :)

Reply:
On `n_jobs` - good catch. No particular reason. My main focus was to make the thing work and refactor it. `n_jobs` should not be defined inside these functions anyways as you said, they should be in a config file, so one can easily change and experiment. Will add back for now as it was before.